### PR TITLE
Addresses #2895, cannot add window frame and divider to skylights

### LINKED ISSUE #2895: Cannot add window frame and divider to skylights
OpenStudio currently [limits the attachment](https://github.com/NREL/OpenStudio/blob/101472e2ec1d04a7d35a94fa261cba8c6f6b9711/openstudiocore/src/model/SubSurface.cpp#L465) of WindowPropertyFrameAndDivider objects to FixedWindow, OperableWindow, and GlassDoor SubSurfaces.  This may have been a limitation of EnergyPlus at some point, but [does not appear to be one now](https://bigladdersoftware.com/epx/docs/8-8/input-output-reference/group-thermal-zone-description-geometry.html#windowpropertyframeanddivider).  The DOE prototype buildings use frames on skylights, which the current implementation prevents.

Solution:
Extend the list of acceptable SubSurface types to include Skylight.  Perhaps it should also include other types as well?

### PR DESCRIPTION
**Pull request by joseph-robertson**
Pull request overview
---------------------

 - Fixes #2895

Please read [OpenStudio Pull Requests](https://github.com/NREL/OpenStudio/wiki/OpenStudio-Pull-Requests) to better understand the OpenStudio Pull Request protocol.

### Pull Request Author

Add to this list or remove from it as applicable.  This is a simple templated set of guidelines.

 - [ ] Model API Changes / Additions
 - [ ] Any new or modified fields have been implemented in the EnergyPlus ForwardTranslator (and ReverseTranslator as appropriate)
 - [ ] Model API methods are tested (in `src/model/test`)
 - [ ] EnergyPlus ForwardTranslator Tests (in `src/energyplus/Test`)
 - [ ] If a new object or method, added a test in NREL/OpenStudio-resources: Add Link
 - [ ] If needed, added VersionTranslation rules for the objects (`src/osversion/VersionTranslator.cpp`)
 - [ ] Checked behavior in OpenStudioApplication, adjusted policies as needed (`src/openstudio_lib/library/OpenStudioPolicy.xml`)
 - [ ] Verified that C# bindings built fine on Windows, partial classes used as needed, etc.
 - [ ] All new and existing tests passes
 - [ ] If methods have been deprecated, update rest of code to use the new methods

**Labels:**

 - [ ] If change to an IDD file, add the label `IDDChange`
 - [ ] If breaking existing API, add the label `APIChange`
 - [ ] If deemed ready, add label `Pull Request - Ready for CI` so that CI builds your PR

### Review Checklist

This will not be exhaustively relevant to every PR.
 - [ ] Perform a Code Review on GitHub
 - [ ] Code Style, strip trailing whitespace, etc.
 - [ ] All related changes have been implemented: model changes, model tests, FT changes, FT tests, VersionTranslation, OS App
 - [ ] Labeling is ok
 - [ ] If defect, verify by running develop branch and reproducing defect, then running PR and reproducing fix
 - [ ] If feature, test running new feature, try creative ways to break it
 - [ ] CI status: all green or justified

### CODE REVIEW

**Comment by jmarrec** [comment]
Seems like the FT should be fine with it too: https://github.com/NREL/OpenStudio/blob/93f1c744ee940afd0e7342b3a857a7a0be2adc81/src/energyplus/ForwardTranslator/ForwardTranslateSubSurface.cpp#L139-L146

(I think instead of always translating the WindowProperty:FrameAndDivider the subsurface should be responsible to call its translation, but that's another story)

**Comment by joseph-robertson** [comment]
It doesn't seem like this ft test for subsurface is very effective: https://github.com/NREL/OpenStudio/blob/develop/src/energyplus/Test/SubSurface_GTest.cpp#L83-L86

**Comment by jmarrec** [comment]
> It doesn't seem like this ft test for subsurface is very effective: https://github.com/NREL/OpenStudio/blob/develop/src/energyplus/Test/SubSurface_GTest.cpp#L83-L86

:) that's the least we can say :D

https://github.com/NREL/OpenStudio/blob/93f1c744ee940afd0e7342b3a857a7a0be2adc81/src/energyplus/Test/SubSurface_GTest.cpp#L83-L86

**Comment by joseph-robertson** [comment]
@jmarrec I filled out the ft test. You want to look it over?

File: src/energyplus/Test/SubSurface_GTest.cpp [hunk 1]
```diff
@@ -83,6 +89,44 @@ TEST_F(EnergyPlusFixture,ReverseTranslator_GlassDoorToSubSurface) {
...
+  Workspace workspace = forwardTranslator.translateModel(model);
+
+  ASSERT_EQ(1u, workspace.getObjectsByType(IddObjectType::FenestrationSurface_Detailed).size());
+  ASSERT_EQ(1u, workspace.getObjectsByType(IddObjectType::WindowProperty_FrameAndDivider).size());
+
+  WorkspaceObject subSurfaceObject = workspace.getObjectsByType(IddObjectType::FenestrationSurface_Detailed)[0];
+  WorkspaceObject frameObject = workspace.getObjectsByType(IddObjectType::WindowProperty_FrameAndDivider)[0];
+
+  ASSERT_TRUE(subSurfaceObject.getTarget(FenestrationSurface_DetailedFields::FrameandDividerName));
+  EXPECT_EQ(frameObject.handle(), subSurfaceObject.getTarget(FenestrationSurface_DetailedFields::FrameandDividerName)->handle());
```
Comment (jmarrec) [inline src/energyplus/Test/SubSurface_GTest.cpp:129]: FT test checks that it is correctly mapped.

**Pull request merged**